### ISSUE TITLE
Implement SmartTab behavior similar to EsLint’s `no-mixed-spaces-and-tabs: smart-tabs`

### ISSUE THREAD

**Issue by maximal** [vuejs/eslint-plugin-vue#2288]
I try to use [SmartTabs](https://www.emacswiki.org/emacs/SmartTabs) approach, and faced an issue with `eslint-plugin-vue` which is successfully resolved in EsLint `indent` basics.

**What category should the rule belong to?**
- [x] Enforces code style (layout)
- [ ] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

My `.eslintrc`:
```yaml
extends:
  - eslint:recommended
  - plugin:vue/vue3-recommended
  - '@vue/eslint-config-typescript'
  - '@vue/eslint-config-prettier/skip-formatting'

# ... ... ...

rules:
  indent:
    - error
    - tab
    - SwitchCase: 1
  no-mixed-spaces-and-tabs:
    - error
    - smart-tabs

# ... ... ...

  vue/html-indent:
    - error # `off` for now, until `smart-tabs` implemented
    - tab
    - alignAttributesVertically: true
  vue/script-indent:
    - error
    - tab
    - switchCase: 1
  vue/first-attribute-linebreak: off
```

```vue
<template>
	<section>
		<template v-if="model.merged_at">
			<!-- GOOD: indent with tabs and alignment with spaces -->
			<i class="bi bi-sign-merge-left"
			   title="Merged" />
		</template>
	</section>
</template>
```

```vue
<template>
	<section>
		<template v-if="model.merged_at">
			<!-- BAD -->
			<i class="bi bi-sign-merge-left"
				title="Merged" />
		</template>
	</section>
</template>
```

Now I have:
<img width="629" alt="image" src="https://github.com/vuejs/eslint-plugin-vue/assets/980679/84d2e466-dc1a-4d8a-9c00-5411ebd43524">

Current rules want me to do that (but that is not what I need by my code style):
<img width="447" alt="image" src="https://github.com/vuejs/eslint-plugin-vue/assets/980679/7e9ebf0d-0c80-4c14-b4cb-75b97694a2e8">

* * *

EsLint `no-mixed-spaces-and-tabs` rule for reference:
https://eslint.org/docs/latest/rules/no-mixed-spaces-and-tabs
